applications/matter_weather_station/thingy53_nrf5340_cpuapp.overlay

```diff
@@ -4,6 +4,64 @@
  * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
  */
 
+/*
+ * This file is used to enable a partition which is used through the flash API
+ * but which is in reality stored in RAM. This is done to load data from
+ * external flash to the network core. Since the network core cannot access
+ * the external flash directly, the data is buffered in RAM before being
+ * copied by the network core to the network core flash.
+ *
+ * The partition must be enabled by the application configuration to prevent
+ * bus faults when mcumgr tries to access simulated partition with network core
+ * image data.
+ */
+
+/ {
+	soc {
+		/* Add a flash controller which has the compatible
+		 * 'zephyr,sim-flash'. This will ensure that the flash
+		 * simulator can use it. None of the other properties in this
+		 * node is used for anything.
+		 */
+		nordic_ram_flash_controller: nordic_ram-flash-controller@0 {
+			compatible = "zephyr,sim-flash";
+			reg = <0x00000000 DT_SIZE_K(40)>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+			erase-value = <0xff>;
+			label = "nordic_ram_flash_flash_controller";
+
+			/* This node label must match that used in the flash
+			 * simulator.
+			 */
+			flash_sim0: flash_sim@0 {
+				status = "okay";
+				compatible = "soc-nv-flash";
+				label = "simulated_flash";
+				erase-block-size = <4096>;
+				write-block-size = <4>;
+				reg = <0x00000000 DT_SIZE_K(256)>;
```

Comment:
`DT_SIZE_K(256)`? and later `reg = <0x00000000 0x00000A000>;`?
Only the `reg = <0x00000000 DT_SIZE_K(40)>;` match 40k that netcore uses. What the other sizes mean?

I know that this is a copy of the already existing overlay and values are probably ignored but can somebody explain these?

Reply:
Please do not fix anything if things work. I just want to know that somebody understands why it works even with these values.

Reply:
Hi @de-nordic , do you have a clue why these values?

Reply:
I suspect that there is copy error. Unfortunately this may have consequences, because partition definitions are dumb and they do not know anything about device, so you may have partition bigger than the device.
It there are calls that would attempt to do some read/write as an offset from the end of partition they may fail, because they will try to calculate that offset using  the partition definition not the real device size, so values passed to device API call will be out of range.

Reply:
@de-nordic, do we want to introduce some changes before 1.7 in that case?

Reply:
I guess that as far as we alloc the amount we require in pm_static it should be fine.... @MarekPieta do we?

Reply:
I hope it is not `A000` as defined in DTS partition.

Reply:
I checked that we use `flash_map` defined in `subsys/partition_manager/flash_map_partition_manager.c`.
In that case, sizes should be defined by pm_static and we should not observe issues.